pkg/proxy/proxy.go

```diff
@@ -113,7 +118,7 @@ func (s *SQLServer) onConn(ctx context.Context, conn net.Conn) {
 	tcpKeepAlive := s.mu.tcpKeepAlive
 
 	// 'maxConns == 0' => unlimited connections
-	if maxConns != 0 && maxConns <= conns {
+	if maxConns != 0 && maxConns < conns {
```

Comment:
It should be `maxConns <= conns` because `s.mu.clients[connID] = clientConn` is not executed yet.